### PR TITLE
Update tools and benchmark datasets to read compressed json

### PR DESCRIPTION
**Pull request by moh-osman3**
### What does this PR do?
This PR
- updates the [logs|traces|metrics] datasets to read files that are in a compressed json format. Files are read line by line. This logic is similar to how the otel contrib filereceiver component will read compressed json.
- updates the [logs|traces|metrics] test data generating tools to write compressed json format. Telemetry is written one at a time followed by a newline. This logic is similar to how the otel contrib fileexporter will write compressed json.

### CODE REVIEW

File: pkg/benchmark/dataset/real_logs_dataset.go [hunk 1]
```diff
@@ -38,24 +42,73 @@ type logUnit struct {
...
+				if errors.Is(err, io.EOF) {
+					return logs, nil
+				}
+				return logs, err
+			}
+		}
+	}
+}
+
 // NewRealLogsDataset creates a new RealLogsDataset from a binary file.
```
Comment (lquerel) [inline pkg/benchmark/dataset/real_logs_dataset.go:76]: Will be good to adjust the comment as the nature of the input file is no longer a binary proto file. Same comment for the other OTLP entities (metrics and traces).

File: tools/logs_benchmark/main.go [hunk 2]
```diff
@@ -91,9 +91,8 @@ func main() {
 		// Build dataset from CSV file or from OTLP protobuf file
 		if strings.HasSuffix(inputFile, ".csv") {
 			ds = CsvToLogsDataset(inputFile)
-		} else if strings.HasSuffix(inputFile, ".pb") {
-			// ToDo Remove
-			rds := dataset.NewRealLogsDataset(inputFiles[i])
+		} else if strings.HasSuffix(inputFile, ".json") {
```
Comment (lquerel) [inline tools/logs_benchmark/main.go:94]: I still need to be able to support protobuf files, so the JSON input format should be in addition to the other formats already supported.

File: tools/logs_benchmark/main.go [hunk 3]
```diff
@@ -64,9 +64,9 @@ func main() {
 	// Define default input file
 	inputFiles := flag.Args()
 	if len(inputFiles) == 0 {
-		println("\nNo input file specified, using default file ./data/otlp_logs.pb")
-		println("CSV and OTLP protobuf files are supported as input files (ext .csv or .pb)")
-		inputFiles = append(inputFiles, "./data/otlp_logs.pb")
+		println("\nNo input file specified, using default file ./data/otlp_logs.json")
```
Comment (lquerel) [inline tools/logs_benchmark/main.go:67]: I'd like to keep the protobuf format the default one (as I'm still using it). The `.json` file extension can be used to detect the format and use the correct reader.

File: tools/logs_gen/main.go [hunk 4]
```diff
@@ -20,20 +20,22 @@ package main
...
+	"github.com/klauspost/compress/zstd"
 	"go.opentelemetry.io/collector/pdata/plog/plogotlp"
 
 	"github.com/f5/otel-arrow-adapter/pkg/datagen"
 )
 
 var help = flag.Bool("help", false, "Show help")
-var outputFile = "./data/otlp_logs.pb"
-var batchSize = 100000
+var outputFile = "./data/otlp_logs.json"
```
Comment (lquerel) [inline tools/logs_gen/main.go:37]: I'd like to keep the ability to generate logs/metrics/traces in binary format (proto). I'm ok to add an option to generate a json version with zstd compression.

**Review by lquerel — CHANGES_REQUESTED** [review]
First of all, thanks for adding the support for the JSON format. But I'd like to keep the option to generate and read metrics/logs/traces data in a binary format. The main reason is that I have a bunch of data already used in various benchmarks and I need to keep them in order to reproduce them. 

The JSON+ZSTD should be integrated as an additional supported format and not as a replacement for the binary format.

File: pkg/benchmark/dataset/real_logs_dataset.go (hunk 1 above)
Comment (moh-osman3) [reply pkg/benchmark/dataset/real_logs_dataset.go]: Thanks for the review! Updated the comments

File: tools/logs_benchmark/main.go (hunk 2 above)
Comment (moh-osman3) [reply tools/logs_benchmark/main.go]: Now supporting both json and the original proto format

File: tools/logs_benchmark/main.go (hunk 3 above)
Comment (moh-osman3) [reply tools/logs_benchmark/main.go]: protobuf is now the default, so all the tools in this repo should work exactly as it was before this change. But now to access the compressed json tooling you must specify a `-format json` option

File: tools/logs_gen/main.go (hunk 4 above)
Comment (moh-osman3) [reply tools/logs_gen/main.go]: Fixed!

File: tools/logs_benchmark/main.go (hunk 2 above)
Comment (lquerel) [reply tools/logs_benchmark/main.go]: Thanks

**Review by lquerel — APPROVED** [review]

**Comment by moh-osman3** [comment]
> @moh-osman3 there is an issue in the build process preventing us to merge this PR. Could you fix it and I will merge this PR. Thanks.

@lquerel Ahh forgot to update one of the test files with the updated function signature. I think it should be fixed now. Thank you!

**Comment by lquerel** [comment]
@moh-osman3 thanks for this PR.

**Pull request merged**